### ISSUE TITLE
julia crashes on sort(data, cols)

### ISSUE THREAD

**Issue by jbrea** [JuliaData/CSV.jl#790]
Julia 1.5.3 crashes when I run the following code
```julia
using Pkg
const TMPDIR = mktempdir()
cd(TMPDIR)
Pkg.activate(".")
Pkg.add("CSV")
Pkg.add("DataFrames")
using CSV, DataFrames
n = 18
d = DataFrame(a = rand(["bla", "bli"], n), b = rand(1:10, n), c = rand(n))
sort(d, ["a", "b"])                         # works fine
CSV.write("test.csv", d)
d2 = CSV.read("test.csv", DataFrame) 
sort(d2, ["a", "b"])                       # works fine if n < 18 or d2 = CSV.read("test.csv", DataFrame, pool = false) 
```
error message
```julia
Illegal inttoptr
	 %23 = ptrtoint %jl_value_t addrspace(10)* %16 to i64, !dbg !49
Illegal inttoptr
	 %24 = inttoptr i64 %23 to i64 addrspace(13)*, !dbg !49
Illegal inttoptr
	 %42 = ptrtoint i64 addrspace(13)* %39 to i64, !dbg !13
Illegal inttoptr
	 %43 = inttoptr i64 %42 to %jl_value_t addrspace(10)*, !dbg !13

signal (6): Aborted
in expression starting at REPL[26]:1
gsignal at /usr/bin/../lib/libc.so.6 (unknown line)
abort at /usr/bin/../lib/libc.so.6 (unknown line)
unknown function (ip: 0x7f05b5c092c9)
_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE at /usr/bin/../lib/libLLVM-10.so (unknown line)
_ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE at /usr/bin/../lib/libLLVM-10.so (unknown line)
_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE at /usr/bin/../lib/libLLVM-10.so (unknown line)
unknown function (ip: 0x7f05b5d1069b)
unknown function (ip: 0x7f05b5d1359b)
unknown function (ip: 0x7f05b5d152c8)
unknown function (ip: 0x7f05b5d15fb2)
unknown function (ip: 0x7f05b5d17198)
unknown function (ip: 0x7f05b5c978ef)
jl_apply_generic at /usr/bin/../lib/libjulia.so.1 (unknown line)
#sortperm#612 at /home/j/.julia/packages/DataFrames/Zx5mm/src/abstractdataframe/sort.jl:487
unknown function (ip: 0x7f0574e10469)
sortperm##kw at /home/j/.julia/packages/DataFrames/Zx5mm/src/abstractdataframe/sort.jl:476 [inlined]
#sort#611 at /home/j/.julia/packages/DataFrames/Zx5mm/src/abstractdataframe/sort.jl:411 [inlined]
sort at /home/j/.julia/packages/DataFrames/Zx5mm/src/abstractdataframe/sort.jl:411
unknown function (ip: 0x7f0574e0b08e)
unknown function (ip: 0x7f05b5cae735)
unknown function (ip: 0x7f05b5cae3be)
unknown function (ip: 0x7f05b5caef00)
unknown function (ip: 0x7f05b5caf9b0)
unknown function (ip: 0x7f05b5ccbcf1)
unknown function (ip: 0x7f05b5ccc398)
jl_toplevel_eval_in at /usr/bin/../lib/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f05a51ad2d1)
unknown function (ip: 0x7f05a4c88dae)
unknown function (ip: 0x7f05a4c899e0)
unknown function (ip: 0x7f05a4c89b2a)
unknown function (ip: 0x7f05a4ca9927)
run_repl at /build/julia/src/julia-1.5.3/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:288
unknown function (ip: 0x7f05a4ea45df)
unknown function (ip: 0x7f05a4ea46b8)
unknown function (ip: 0x7f05b5ca53e4)
jl_f__apply_latest at /usr/bin/../lib/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f05a4ef6987)
unknown function (ip: 0x7f05a4f02437)
unknown function (ip: 0x7f05a4f0297e)
unknown function (ip: 0x7f05a4f02ad5)
unknown function (ip: 0x55cc0d7624fe)
unknown function (ip: 0x55cc0d7620a7)
__libc_start_main at /usr/bin/../lib/libc.so.6 (unknown line)
unknown function (ip: 0x55cc0d76215d)
Allocations: 69700642 (Pool: 69675400; Big: 25242); GC: 73
zsh: abort (core dumped)  julia
```

**Comment by jbrea** [comment]
Related: if the string column contains a missing entry, `pool = false` doesn't help. I.e. replace the last 5 lines above by
```julia
d = DataFrame(a = [rand(["bla", "bli"], n); missing], b = rand(1:10, n+1), c = rand(n+1))
sort(d, ["a", "b"])                         # works fine
CSV.write("test.csv", d)
d2 = CSV.read("test.csv", DataFrame, pool = false) 
sort(d2, ["a", "b"])                     # crashes.
```

**Comment by quinnj** [comment]
I don't see any crash on 1.5.2 or pretty recent Julia master on my macOS. Could you share what platform you're on? 32-bit vs. 64-bit?

Anyone else out there that can reproduce?

**Comment by jbrea** [comment]
```
julia> versioninfo()
Julia Version 1.5.3
Commit 788b2c77c1* (2020-11-09 13:37 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-5600U CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-10.0.1 (ORCJIT, broadwell)
```

**Comment by quinnj** [comment]
Hmmm, I don't see a crash on:
```julia
julia> versioninfo()
Julia Version 1.5.3
Commit 788b2c77c1* (2020-11-08 13:37 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Atom(TM) CPU N455  @ 1.66GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, bonnell)
```

It may be something that is already fix on the soon-to-be-released 1.6 branch; could you try https://julialangnightlies-s3.julialang.org/bin/linux/x64/julia-latest-linux64.tar.gz and see if you still get the crash?

**Comment by jbrea** [comment]
Thanks for looking into this. Indeed, I don't get any crash with
```julia
julia> versioninfo()
Julia Version 1.7.0-DEV.29
Commit d8975fab04* (2020-12-11 04:35 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-5600U CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.0 (ORCJIT, broadwell)
```
So it may be an unhappy coincidence that it crashes under Julia 1.5.3 with libLLVM-10.0.1.
